### PR TITLE
Fix for lirc message spam

### PR DESCRIPTION
**Pull request by Paxxi**
I reworked the LIRC code to match the one in IRSS for Windows and removed the calls to Init in our slow loop. New behvaiour is that it will try to connect 60 times at launch, every 5 seconds, after that it gives up and won't be used.
I've tried this on OpenElec and it works fine with a remote at least.

### CODE REVIEW

**Comment by Paxxi** [comment]
@mkortstiege and @garbear what do you think?

**Comment by mkortstiege** [comment]
Code is lacking an enabled check which makes it impossible to start without lirc. Other than that it looks good and fixes the issue. Thanks.

Btw, I think retrying for 5 minutes is a bit long. What about max 3 minutes?

**Comment by Paxxi** [comment]
Do you mean when compiled without lirc support?

**Comment by mkortstiege** [comment]
No, starting with `-n` or `--nolirc`. This is calling CInputManager::Get().DisableRemoteControl(); in xbmc/AppParamParser.cpp but has no effect currently.

**Comment by Paxxi** [comment]
Ahh had missed that it was called there, but that option is going away with @topfs2 apparat pr afaik

**Comment by mkortstiege** [comment]
Well, than this a non-issue :)

**Comment by mkortstiege** [comment]
jenkins build this please

**Comment by mkortstiege** [comment]
@garbear objections or comments?

File: xbmc/input/linux/LIRC.cpp [hunk 1]
```diff
@@ -109,109 +101,55 @@ void CRemoteControl::Disconnect()
...
-            CLog::Log(LOGERROR, "LIRC %s: fdopen failed: %s", __FUNCTION__, strerror(errno));
-        }
-        else
-          CLog::Log(LOGERROR, "LIRC %s: fcntl(F_SETFL) failed: %s", __FUNCTION__, strerror(errno));
-      }
-      else
-        CLog::Log(LOGERROR, "LIRC %s: fcntl(F_GETFL) failed: %s", __FUNCTION__, strerror(errno));
-    }
-    else
+    if (XbmcThreads::SystemClockMillis() - time >= iMsRetryDelay)
```
Comment (garbear) [inline xbmc/input/linux/LIRC.cpp]: Can you use AbortableWait() as in https://github.com/garbear/xbmc/commit/65b96f2#diff-79dbb285338e176975ab76f60ad0a4e1R208?

This way we avoid additional context switches every second and don't hang for up to a second after Disconnect() is called

Comment (Paxxi) [inline xbmc/input/linux/LIRC.cpp]: Good catch, updating both lirc and irss versions to use AbortableWait instead.

**Comment by Paxxi** [comment]
jenkins build this please

**Comment by Paxxi** [comment]
Jenkins build this please again

**Comment by Paxxi** [comment]
@garbear yup, me confusing events and stopwatch, fixing this now

**Comment by Paxxi** [comment]
jenkins build and merge

**Comment by mkortstiege** [comment]
looks like a jenkins hiccup. jenkins build and merge

**Comment by MartijnKaijser** [comment]
Guess it's stuck on Addons perhaps. @Memphiz ping before we merge

**Comment by Memphiz** [comment]
the jenkins merge trigger doesn't work as long as the builds are unstable (meaning that binary addons didn't build) - can't do anything about it ...

**Pull request merged**